src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -49,6 +52,75 @@ public static Encoding OutputEncoding
             get { return GetConsoleEncoding(); }
         }
 
+        private static readonly object InternalSyncObject = new object();
+        private static SyncTextReader s_stdInReader;
+        private const int DefaultBufferSize = 255;
+
+        public static SyncTextReader StdInReader
+        {
+            get
+            {
+                if (Console.IsInputRedirected)
+                    throw new InvalidOperationException();
+
+                SyncTextReader result = Volatile.Read(ref s_stdInReader);
+
+                if (result == null)
+                {
+                    lock (InternalSyncObject)
+                    {
+                        result = Volatile.Read(ref s_stdInReader);
+                        if (result == null)
+                        {
+                            result = SyncTextReader.GetSynchronizedTextReader(
+                            new StdInStreamReader(
+                                stream: OpenStandardInput(),
+                                encoding: InputEncoding,
+                                detectEncodingFromByteOrderMarks: false,
+                                bufferSize: DefaultBufferSize,
+                                leaveOpen: true));
+                            Volatile.Write(ref s_stdInReader, result);
+                        }
+                    }
+                }
+
+                return s_stdInReader;
+            }
+        }
+
+        private const int DefaultConsoleBufferSize = 256; // default size of buffer used in stream readers/writers
+        public static TextReader In
```

Comment:
Sure, but every time someone calls this, it's potentially going to return a new instance.  Typically properties aren't supposed to return new instances every time you call them.  So it seems this would be better as a method named something like GetOrCreateReader.